lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -2336,6 +2336,305 @@ class ConvertCastEquivalentOp : public OpConversionPattern<OpTy> {
+                  Value H = b.create<linalg::IndexOp>(loc, 2);
+                  Value W = b.create<linalg::IndexOp>(loc, 3);
+                  Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+                  SmallVector<Value> gridIndex_0{N, H, W, zero};
+                  Value lw_0 =
+                      b.create<tensor::ExtractOp>(loc, lower_0, gridIndex_0);
+                  Value lw_1 =
+                      b.create<tensor::ExtractOp>(loc, lower_1, gridIndex_0);
+                  Value lw_0a =
+                      b.create<arith::IndexCastOp>(loc, b.getIndexType(), lw_0);
+                  Value lw_1a =
+                      b.create<arith::IndexCastOp>(loc, b.getIndexType(), lw_1);
+                  SmallVector<Value> inputIndex00{N, C, lw_0a, lw_1a};
+                  Value result_00 = rewriter.create<tensor::ExtractOp>(
+                      loc, input, inputIndex00);
+                  Value up_1 =
+                      b.create<tensor::ExtractOp>(loc, upper_1, gridIndex_0);
+                  Value up_1a =
+                      b.create<arith::IndexCastOp>(loc, b.getIndexType(), up_1);
+                  SmallVector<Value> inputIndex01{N, C, lw_0a, up_1a};
+                  Value result_01 = rewriter.create<tensor::ExtractOp>(
+                      loc, input, inputIndex01);
+                  Value nv_1 =
+                      b.create<tensor::ExtractOp>(loc, notValid_1, gridIndex_0);
+                  Value zeroVal = rewriter.create<arith::ConstantOp>(
+                      loc, rewriter.getFloatAttr(floatType, 0.0));
+                  Value result_01a = rewriter.create<arith::SelectOp>(
+                      loc, nv_1, zeroVal, result_01);
+                  Value up_0 =
+                      b.create<tensor::ExtractOp>(loc, upper_0, gridIndex_0);
+                  Value up_0a =
+                      b.create<arith::IndexCastOp>(loc, b.getIndexType(), up_0);
+                  SmallVector<Value> inputIndex10{N, C, up_0a, lw_1a};
+                  Value result_10 = rewriter.create<tensor::ExtractOp>(
+                      loc, input, inputIndex10);
+                  Value nv_0 =
+                      b.create<tensor::ExtractOp>(loc, notValid_0, gridIndex_0);
+                  Value result_10a = rewriter.create<arith::SelectOp>(
+                      loc, nv_0, zeroVal, result_10);
+                  SmallVector<Value> inputIndex11{N, C, up_0a, up_1a};
+                  Value result_11 = rewriter.create<tensor::ExtractOp>(
+                      loc, input, inputIndex11);
+                  Value result_11a = rewriter.create<arith::SelectOp>(
+                      loc, nv_0, zeroVal, result_11);
+                  Value result_11b = rewriter.create<arith::SelectOp>(
+                      loc, nv_1, zeroVal, result_11a);
+
+                  Value d_00 =
+                      b.create<tensor::ExtractOp>(loc, delta_00, gridIndex_0);
```

Comment:
Again, these extractions should be performed via the `linalg.Generic`'s operands.

Reply:
done